File: swarmfiles/makeswarmfile.jl
using DelimitedFiles

datafiles = readdir("/Users/carsonc/github/covid-sicr/data")
datafiles = readdir("/Users/carsonc/github/VaxBRAModelEPI/USStates/")

datafiles = datafiles[occursin.("covidtimeseries_US_",datafiles)]

regions = Array{String,1}(undef,0)
for datafile in datafiles
    push!(regions,replace(replace(datafile,".csv"=>""),"covidtimeseries_"=>""))
end

# models = ["fulllinearmodel","reducedlinearmodelq0","reducedlinearmodelq0ctime","reducedlinearmodelNegBinom","nonlinearmodelq0ctime","nonlinearmodel"]
# models = ["nonlinearmodelq0ctime"]
# models = ["nonlinearmodel"]
# models = ["linearm2R","linearmqc2R"]
models = ["SICRM","SICRLM","SICRMQC","SICRLMQC","SICRM2R","SICRMQC2R","SICRLM2R","SICRLMQC2R"]
models = ["SICRMQ","SICRLMQ","SICRMQ2R","SICRLMQ2R"]
models = ["SICRMQC2R2DX2"]
models = ["SICR2HV"]

for modelname in models

    iter = "-it=10000"
    warm = "-wm=6000"
    format = "-f=1"
    fitpath = "-fp='/data/carsonc/covid-19/fitUS06-27-20V/'"
    modelpath = "-mp='/home/carsonc/covid-sicr/models/'"
    # datapath = "-dp='/home/carsonc/covid-sicr/data/'"
    datapath = "-dp='/home/carsonc/VaxBRAModelEPI/USStates/'"
    ad = "-ad=.85"
    ft = "-ft=1"
    ld = "-ld='06/27/20'"
    runfile = "&& python /home/carsonc/covid-sicr/scripts/runV.py"


    swarmname = modelname * ".swarm"

    f = open(swarmname,"w")

    for region in regions
        # init = "-i='/data/carsonc/covid-19/fit10/" * modelname * "_" * region * ".pkl'"
        init = ""
        regionarg = "-r='" * region * "'"
        writedlm(f,["source /data/carsonc/conda/etc/profile.d/conda.sh \\" ])
        writedlm(f,[ "&& conda activate covid-19 \\" ])
        writedlm(f,[ runfile modelname regionarg modelpath datapath fitpath iter warm format ad ft ld])
        # writedlm(f,[ "/data/carsonc/conda/envs/covid-19/bin/python run.py" modelname region fitpath iter warm format ])
    end

    close(f)
end

File: swarmfiles/makeswarmfile.py
import glob
import os

# Create list of ROIs we want that are found in data_path
data_path = '../data/'
rois = []

# get US rois
for name in glob.glob(data_path + 'covidtimeseries_US_??.csv'):
    roi = os.path.basename(name)
    rois.append(roi[16:-4])

# add Canadian provinces
canadian_provinces = ['Alberta', 'BC', 'Manitoba', 'New Brunswick', 'NL',
                'Nova Scotia', 'Nunavut', 'NWT', 'Ontario', 'PEI', 'Quebec',
                'Saskatchewan', 'Yukon']
for prov in canadian_provinces:
    prov = 'CA_' + prov
    rois.append(prov)

rois.sort()

models = ['SICRdiscrete', 'SICRdiscrete1', 'SICRMQC2R2DX']


for model in models:
    swarmFile = open(f"../{model}.swarm", "w")
    for roi in rois:
        line = ("source /data/schwartzao/conda/etc/profile.d/conda.sh "
            "&& conda activate covid "
            f"&& python /home/schwartzao/covid-sicr/scripts/run.py {model} "
            f"-r='{roi}' "
            "-mp='/home/schwartzao/covid-sicr/models/' "
            "-fp='/data/schwartzao/covid-sicr/fits20210105/' " # changes
            "-it=10000	-wm=6000	-f=1	-ad=.85	-ft=1"
            )
        swarmFile.write(line)
        swarmFile.write('\n')
    swarmFile.close()
